src/common/transformations/src/transformations/op_conversions/einsum_decomposition.cpp

```diff
@@ -174,7 +170,118 @@ void update_operands(ov::OutputVector& input_nodes,
     input_subscripts.erase(input_subscripts.begin() + input_ind1);
     input_subscripts.push_back(new_subscript);
 }
+using LabelDimMap = std::unordered_map<std::string, std::vector<size_t>>;
+
+/// \brief Computes a mapping from labels to dimensions based on the input rank and subscript.
+///
+/// This function processes the input subscript to extract labels and maps them to the corresponding
+/// dimensions of the input tensor. The function also considers the presence of ellipsis ("...") in
+/// the labels and adjusts the dimension map accordingly.
+///
+/// \param input_rank The rank of the input tensor. It can be static or dynamic.
+/// \param input_subscript The subscript string representing the labels of the input tensor dimensions.
+/// \return A map where the keys are labels (strings) and the values are vectors of dimension indices.
+///
+LabelDimMap compute_label_dim_map(const ov::Rank& input_rank, const std::string& input_subscript) {
+    static const std::string ellipsis = "...";
+    const auto labels = ov::op::v7::Einsum::extract_labels(input_subscript);
+    const auto static_input_rank = input_rank.is_static();
+    OPENVINO_ASSERT(static_input_rank || (std::find(labels.begin(), labels.end(), ellipsis) == labels.end()),
+                    "Input rank cannot be dynamic in case of ellipsis in input subscript");
+    const size_t input_rank_length = static_input_rank ? input_rank.get_length() : labels.size();
+    OPENVINO_ASSERT(input_rank_length >= labels.size());
+    const size_t num_broadcasted_dims = input_rank_length - labels.size() + 1;
```

Comment:
why does it add `+1`? I think it is valid only if ellipsis exists but ellipsis can be absent. Please check